### PR TITLE
Added a method to check if user paid for the app when the app was downloaded.

### PR DESCRIPTION
**Pull request by wootwoot1234**

### CODE REVIEW

**Comment by marcgreenstock** [comment]
Thanks for contributing!

Implementation of `SKReceiptRefreshRequest` is something definitely something that should be implemented on this project. But it can be used for more than just checking if an app is paid for or not.

It's main use (as I understand) is for refreshing a receipt if a connection is interrupted while making an in-app purchase or in the case of `appStoreURL` not responding correctly.

I suggest changing the js method name `paidForApp` to be less specific to your use case and having it return the specific error on fail in the callback.

**Comment by chirag04** [comment]
agree with @marcgreenstock.

**Comment by wootwoot1234** [comment]
@marcgreenstock @chirag04,

Unfortunately the code I submitted is on the edge of or past my understanding of objective-c.  I would make the changes you're talking about to contribute to this project but I don't know how and this code does what I needed.  Maybe someone else can make the changes?

**Comment by marcgreenstock** [comment]
@wootwoot1234 no dramas, i'm a noob at objective-c too, i'll fork your branch and make it fit with my suggestions over the next few days.

**Comment by changchup** [comment]
@marcgreenstock  Did you have any success merging this?

File: InAppUtils/InAppUtils.m [hunk 1]
```diff
@@ -189,6 +189,37 @@ - (void)dealloc
     [[SKPaymentQueue defaultQueue] removeTransactionObserver:self];
 }
 
+RCT_EXPORT_METHOD(paidForApp:(RCTResponseSenderBlock)callback)
```
Comment (grabbou) [inline InAppUtils/InAppUtils.m]: This should be called `restorePurchase` and if it returns with some results - it was purchased. In fact every app according to apple guidelines needs that button.

**Comment by chirag04** [comment]
we already have `restorePurchase` supported. Idk we need anything more here.

**Comment by wootwoot1234** [comment]
@chirag04 I'm just coming back to this after working on another project for a while.  This pull request is not another ```restorePurchase``` method, i don't know what @grabbou was talking about.  This allows you to know if the app was paid for or not.  I think it's very helpful for people migrating from or to a freemium model.

I just fixed the conflicts in the readme file.  Should be ready to go if agree to merge it.

File: InAppUtils/InAppUtils.m (hunk 1 above)
Comment (wootwoot1234) [inline InAppUtils/InAppUtils.m]: @grabbou this isn't a ```restorePurchase``` function, that already exists in the project.

File: InAppUtils/InAppUtils.m [hunk 2]
```diff
@@ -199,6 +199,37 @@ - (void)dealloc
...
+        if (callback) {
+            callback(@[@"paid_for_app"]);
+            [_callbacks removeObjectForKey:key];
+        }
+    }
+}
+
+- (void)request:(SKRequest*)request didFailWithError:(NSError *)error{
+    NSLog(@"NO, you need to buy it");
+    NSString *key = RCTKeyForInstance(@"paidForAppRequest");
```
Comment (chirag04) [inline InAppUtils/InAppUtils.m]: This delegate method will catch all request failures. even the ones for buying something. Eventually, you are calling a wrong callback when request fails.

**Comment by chirag04** [comment]
I don't think this is is the right way of doing it.

if you want to know if the user already paid for the app, just call `restorePurchase`. If `restorePurchase` succeeds that means the user paid for it.

**Comment by ncpope** [comment]
Not sure why this isn't getting merged in. Checking for an IAP is different than an up front app purchase. The idea for this functionality is to allow for a pricing model change.

For example: You create your app for $0.99 and at a later point you decide to go freemium where you give away the app for free and sell certain functionality for $0.99. You wouldn't want the people who originally purchased the app to have to now pay the $0.99 again when they update their app.

**Comment by ryancoughlin** [comment]
Running into this same issue. @wootwoot1234 do you have time to rebase and merge?

**Comment by akshetpandey** [comment]
Propose closing this PR.

@chirag04 : I think this PR should be closed as the wrong solution. This is better handled in the JS land by enumerating through the receipts and checking what is needed.

**Comment by chirag04** [comment]
please use restorePurchases.

**Pull request closed without merging**